scipy/sparse/_index.py

```diff
@@ -31,18 +31,28 @@ class IndexMixin:
     """
     def __getitem__(self, key):
         row, col = self._validate_indices(key)
+
+        def vector(data):
```

Comment:
Rather than a "pass-through or fail" function how about something like this:

```python
def check_1d_slice():
    if self._is_array:
        raise NotImplementedError("...etc etc...")
```

And then we can call it prior to doing any work in the 1d slicing cases.

Reply:
I don't see how this logic can easily be replaced by that function.  The signature would probably have to be `def check_1d_ix(row, col)`, and then it'd have to duplicate all the logic already in place.  Also, this is probably not the place to be worried about performance, so first doing some work and then bailing seems fine.

Reply:
Done.